### PR TITLE
28 refactor data access implementation

### PR DESCRIPTION
**Pull request by DanielLivshits**
### Description 
Data Access versions of the entities (except for week) were implemented with the intention of not violating the dependency rule. We have since concluded that having the data access use the entities directly would not actually violate the dependency rule (the dependency points inwards), so we'll refactor the (slightly more complete) version of DataAccessDay into the entities layer and remove the other two entities. Any references to the DataAccess Entities will have to be changed accordingly.   
### Important Notes 
This PR should also refactor some other changes noticed in previously merged files.

### CODE REVIEW

**Review by Fard-Faru — APPROVED** [review]
Thanks for Refactor Daniel!

**Review by POPPININ — APPROVED** [review]
Thanks for removing the redundant data access versions of the entities. The code looks much leaner now.

File: src/main/java/modify_event_use_case/ModifyEventDataAccess.java [hunk 1]
```diff
@@ -53,12 +53,12 @@ public boolean titleExistsInDay(int dayIndex, String title) {
...
     @Override
     public boolean isTimeConflict(int dayIndex, String title, LocalTime newStart, LocalTime newEnd) {
-        HashMap<String, DataAccessEvent> eventsToCheck = days.get(dayIndex).getEvents();
+        HashMap<String, Event> eventsToCheck = days.get(dayIndex).getEvents();
         for(String name: eventsToCheck.keySet()){
             if(!Objects.equals(name, title)){
                 LocalTime start = eventsToCheck.get(name).getStartTime();
                 LocalTime end = eventsToCheck.get(name).getEndTime();
-                if(!(newStart.isAfter(end) || newEnd.isBefore(start))){
+                if(newStart.isBefore(end) &&  newEnd.isAfter(start)){
```
Comment (missionStrawberry) [inline src/main/java/modify_event_use_case/ModifyEventDataAccess.java:61]: Upon further consideration, the if statement should be:
!(startTime.isBefore(end) && (endTime.isBefore(start) || endTime.equals(start))) &&
            !(start.isBefore(endTime) && (end.isBefore(startTime) || end.equals(startTime)))

Comment (missionStrawberry) [reply src/main/java/modify_event_use_case/ModifyEventDataAccess.java:61]: And can you make the same change to CreateEventDataAccess?

**Review by missionStrawberry — CHANGES_REQUESTED** [review]
Some of my comments may be disregarded due to discussions on discord, but there is one change that is important to make in createEvent and modifyEvent data access.

File: src/main/java/entities/Event.java [hunk 2]
```diff
@@ -1,20 +1,23 @@
...
     private LocalTime startTime;
     private LocalTime endTime;
 
-    protected Event(String title, LocalTime startTime, LocalTime endTime) {
+    /**
+     * @param title the title of this event
+     * @param startTime the time that this event starts at
+     * @param endTime the time that this event ends at
+     */
+    public Event(String title, LocalTime startTime, LocalTime endTime) {
```
[the remark this answers is not in the document]
Comment (DanielLivshits) [reply src/main/java/entities/Event.java:20]: it should, but making it protected breaks several implementations that need to be resolved before we can make that change.

File: src/main/java/entities/Task.java [hunk 3]
```diff
@@ -1,17 +1,19 @@
...
+ */
+public class Task implements Serializable {
     private String title;
     private boolean completed;
 
-    protected Task(String title) {
+    /**
+     * @param title the title of this task
+     */
+    public Task(String title) {
```
[the remark this answers is not in the document]
Comment (DanielLivshits) [reply src/main/java/entities/Task.java:16]: see above comment.

File: src/main/java/modify_event_use_case/ModifyEventDataAccess.java (hunk 1 above)
Comment (DanielLivshits) [reply src/main/java/modify_event_use_case/ModifyEventDataAccess.java:61]: the statement you suggested can be simplified a bit due to us checking for chronological order on start and end times.

Comment (DanielLivshits) [reply src/main/java/modify_event_use_case/ModifyEventDataAccess.java:61]: After writing out a derivation, I can affirm that the statement currently in the file and the statement being suggested are logically equivalent.

Comment (missionStrawberry) [reply src/main/java/modify_event_use_case/ModifyEventDataAccess.java:61]: Nevermind it was fine.

**Review by missionStrawberry — APPROVED** [review]
After discussion I am ready to approve!

**Review by Fard-Faru — APPROVED** [review]
Good Work Daniel!

**Pull request merged**